### DISCUSSION TITLE
How to have form with equal width fields without taking full page width

### DISCUSSION THREAD

**Discussion by smartm0use** [fomantic/Fomantic-UI#1306] [Collections (form, grid, menu, message, table, ...)] [answered]
Hello, I'm building form like this:

![Screenshot at 2020-02-03 16-09-43](https://user-images.githubusercontent.com/5399265/73664724-2bb1b680-46a0-11ea-902d-254d7b5a1fa6.png)

As you can see I get different width for every type of input. I have tried to use `equal width` class but i get this:

![Screenshot at 2020-02-03 16-09-22](https://user-images.githubusercontent.com/5399265/73664778-45eb9480-46a0-11ea-9589-e94fde43524e.png)

It could be uncomfortable to use on large screen :(

Is there any clean way to have the following equal width fields form?

![download](https://user-images.githubusercontent.com/5399265/73664915-877c3f80-46a0-11ea-91f9-8986ed63ce77.png)

**Comment by ColinFrick** [comment]
You could use "Evenly Divided" forms https://fomantic-ui.com/collections/form.html#evenly-divided
https://jsfiddle.net/Lu2y5ecd/

**Comment by smartm0use** [comment]
Yes, I see the trick is to use `three` class even in case of having two fields.
Anyway it is not working as I mean: it is still taking the whole width of the page, and this could be not nice on large screen.

**Comment by GammaGames** [comment]
@smartm0use try putting it in a [container](https://fomantic-ui.com/elements/container.html#/definition)

**Comment by smartm0use** [comment]
> @smartm0use try putting it in a [container](https://fomantic-ui.com/elements/container.html#/definition)

Uhm, thank you for your hint... but it is centered in the page...

![download](https://user-images.githubusercontent.com/5399265/73758636-28cfc800-476b-11ea-9ee2-612a6c8ce464.png)

**Comment by GammaGames** [comment, marked as answer]
Huh, I didn't realize you wanted it left aligned. You could just override the `margin-left: auto !important` rule in the container, or use something like a grid instead.

This might be an easy feature request as well, does anyone have any thoughts on a left or right "floated" container? The CSS might look something like:

```css
.ui.container:not(.fluid) {
    width: 933px;
}
.ui.container:not(.left.floated) {
    margin-right: auto;
}
.ui.container:not(.right.floated) {
    margin-left: auto;
}
```

**Comment by smartm0use** [comment]
Yes, it could be a new feature for Fomantic-UI containers.
Thank you for your message.

**Comment by ko2in** [comment]
Closed since the issue seems to be resolved. Please create a separate issue if you've any feature request with that.
